### PR TITLE
Docs [unit-testing]: Add an explanation to the expanding notation

### PR DESCRIPTION
**Pull request by dannystaple**
After some team discussion, we found a useful way to explain the sample generation, and feel it's worth sharing in the docs.

Signed-off-by: Danny Staple <danny@orionrobots.co.uk>

### CODE REVIEW

File: docs/configuration/unit_testing_rules.md [hunk 1]
```diff
@@ -77,14 +77,17 @@ series: <string>
...
 #    '_' represents a missing sample from scrape
 #    'stale' indicates a stale sample
 # Examples:
-#     1. '-2+4x3' becomes '-2 2 6 10'
-#     2. ' 1-2x4' becomes '1 -1 -3 -5 -7'
-#     3. ' 1 _x3 stale' becomes '1 _ _ _ stale'
+#     1. '-2+4x3' becomes '-2 2 6 10' - series starts at -2, then 3 further samples incrementing by 4.
+#     2. ' 1-2x4' becomes '1 -1 -3 -5 -7' - series starts at 1, then 4 further samples decrementing by 2.
+#     3. ' 1x4' becomes '1 1 1 1 1' - shorthand for '1+0x4', series starts at 1, then 4 further samples incrementing by 0.
+#     4. ' 1 _x3 stale' becomes '1 _ _ _ stale' - the missing sample cannot increment, so only 3 samples are produced by the '_x3' expression.
```
Comment (roidelapluie) [inline docs/configuration/unit_testing_rules.md:90]: You mean 2 samples?

Comment (dannystaple) [reply docs/configuration/unit_testing_rules.md:90]: The `_x3` definitely behaves differently - so the `one missing sample plus 2 more` didn't seem correct. It produces a total of 3 samples.
This is in contrast with `1x3`, which will produce a total of 4 samples due to the shorthand expansion that the numeric version can apply.

Comment (roidelapluie) [reply docs/configuration/unit_testing_rules.md:90]: _x3 should not produce samples

Comment (dannystaple) [reply docs/configuration/unit_testing_rules.md:90]: Ah - terminology - good point - let me amend that term.

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**